erpnext/stock/report/stock_balance_tree/stock_balance_tree.py

```diff
@@ -0,0 +1,321 @@
+					tot_open_val = 0
+				        tot_in_val = 0
+				        tot_out_val = 0
+					tot_bal_qty = 0
+					tot_bal_val = 0
+        	                        tot_open_qty = tot_open_qty + rows[4]
+					tot_in_qty = tot_in_qty + rows[7]
+					tot_out_qty = tot_out_qty + rows[10]
+					tot_open_val = tot_open_val + rows[6]
+					tot_in_val = tot_in_val + rows[9]
+					tot_out_val = tot_out_val + rows[12]
+					tot_bal_qty = tot_bal_qty + rows[13]
+					tot_bal_val = tot_bal_val + rows[15]
+					item_group_prev = item_group_work
+                        else:
+					summ_data.append([parent_item_group_prev, " ", " ", 
+				 	" ", ptot_open_qty, 0, ptot_open_val,
+					ptot_in_qty, 0, ptot_in_val, ptot_out_qty, 0, ptot_out_val,
+					ptot_bal_qty, 0, ptot_bal_val				
+ 					])	
+					
+					summ_data.append([parent_item_group_work, item_group_work, rows[2], 
+				 	rows[3], rows[4], rows[5],
+					rows[6], rows[7],
+					rows[8], rows[9], rows[10],
+					rows[11], rows[12], rows[13],
+					rows[14], rows[15]		
+ 					])
+                                
+					tot_open_qty = 0
+					tot_in_qty = 0
+					tot_out_qty = 0
+					tot_open_val = 0
+				        tot_in_val = 0
+				        tot_out_val = 0
+					tot_bal_qty = 0
+					tot_bal_val = 0
+					ptot_open_qty = 0
+					ptot_in_qty = 0
+					ptot_out_qty = 0
+					ptot_open_val = 0
+				        ptot_in_val = 0
+				        ptot_out_val = 0
+					ptot_bal_qty = 0
+					ptot_bal_val = 0
+        	                        tot_open_qty = tot_open_qty + rows[4]
+					tot_in_qty = tot_in_qty + rows[7]
+					tot_out_qty = tot_out_qty + rows[10]
+					tot_open_val = tot_open_val + rows[6]
+					tot_in_val = tot_in_val + rows[9]
+					tot_out_val = tot_out_val + rows[12]
+					tot_bal_qty = tot_bal_qty + rows[13]
+					tot_bal_val = tot_bal_val + rows[15]
+
+					item_group_prev = item_group_work
+					parent_item_group_prev = parent_item_group_work
+
+		item_count = item_count + 1
+
+        summ_data.append([parent_item_group_prev, " ", " ", 
+				 	" ", ptot_open_qty, 0, ptot_open_val,
+					ptot_in_qty, 0, ptot_in_val, ptot_out_qty, 0, ptot_out_val,
+					ptot_bal_qty, 0, ptot_bal_val				
+ 					])  
+              
+		
+		
+		
+		
+						
+	return columns, summ_data
+
+
+
+                   
+
+
+def get_columns():
+	"""return columns"""
+
+	columns = [
+		_("Parent Item Group")+"::100",
+                _("Item Group")+"::100",
+                _("Item Name")+":Link/Item:100",
+		_("Stock UOM")+":Link/UOM:90",
+		_("Opening Qty")+":Float:100",
+                _("Opening Rate")+":Float:100",
+		_("Opening Value")+":Float:100",
+		_("In Qty")+":Float:80",
+		_("In Rate")+":Float:80",
+		_("In Value")+":Float:80",
+		_("Out Qty")+":Float:80",
+		_("Out Rate")+":Float:80",
+		_("Out Value")+":Float:80",
+		_("Closing Qty")+":Float:100",
+		_("Closing Rate")+":Float:90",
+                _("Closing Value")+":Float:90",
+			]
+
+	return columns
+
+def get_conditions(filters):
+	conditions = ""
```

Comment:
this seems similar to the above report. Maybe just import the function from that report instead of copy-paste!